### PR TITLE
refactor(lobby): Convert React components to Typescript

### PR DESCRIPTION
**Pull request by c-w**
Similar to @coyotte508 in https://github.com/boardgameio/boardgame.io/issues/190#issuecomment-617446765, I recently ran into an issue importing `boardgame.io/client` due to the missing types in the Lobby React components. As such, this pull request converts the Lobby React components to Typescript which should fix the import error.

#### Checklist

* [x] Use a separate branch in your local repo (not `master`).
* [x] Test coverage is 100% (or you have a story for why it's ok).

### CODE REVIEW

File: src/lobby/react.tsx [hunk 1]
```diff
@@ -8,20 +8,64 @@
...
+  matchID: string;
+  playerID: string;
+  credentials?: string;
+};
+
+type LobbyProps = {
+  gameComponents: GameComponent[];
+  lobbyServer?: string;
+  gameServer?: string;
+  debug?: boolean;
```
Comment (delucis) [inline src/lobby/react.tsx:38]: This should probably align with the `debug` option that can be passed to the client functions:

https://github.com/boardgameio/boardgame.io/blob/af435611685d72e3ea6e1a81909c9bb6522a18dd/src/client/client.ts#L104

This allows users to pass the full debug panel implementation if they want to include it in production builds ([docs](https://boardgame.io/documentation/#/debugging?id=using-the-debug-panel-in-production)).

File: src/lobby/react.tsx [hunk 2]
```diff
@@ -42,16 +86,7 @@ const LobbyPhases = {
  *   A React component that provides a UI to create, list, join, leave, play or
  *   spectate matches (game instances).
  */
-class Lobby extends React.Component {
-  static propTypes = {
```
Comment (delucis) [inline src/lobby/react.tsx:46]: Could we keep prop types for this public-facing component? Keeping prop types gives some development warnings for developers who aren’t using Typescript, which might be better than nothing for them.

**Review by delucis — CHANGES_REQUESTED** [review]
Nice work! A couple of small questions, but otherwise this looks great.

File: src/lobby/react.tsx (hunk 2 above)
Comment (c-w) [reply src/lobby/react.tsx:46]: Done in 3bacaef.

File: src/lobby/react.tsx (hunk 1 above)
Comment (c-w) [reply src/lobby/react.tsx]: Done in a994a43.

**Review by delucis — APPROVED** [review]

**Comment by delucis** [comment]
Thanks @c-w!

**Pull request merged**